components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/NotificationTemplateManagerImpl.java

```diff
@@ -0,0 +1,720 @@
+            String errorCode =
+                    I18nEmailUtil.prependOperationScenarioToErrorCode(
+                            TemplateMgtConstants.ErrorMessages.ERROR_CODE_EMPTY_TEMPLATE_CHANNEL.getCode(),
+                            TemplateMgtConstants.ErrorScenarios.NOTIFICATION_TEMPLATE_MANAGER);
+            throw new NotificationTemplateManagerClientException(errorCode,
+                    TemplateMgtConstants.ErrorMessages.ERROR_CODE_EMPTY_TEMPLATE_CHANNEL.getMessage());
+        }
+        if (NotificationChannels.SMS_CHANNEL.getChannelType()
+                .equalsIgnoreCase(notificationTemplate.getNotificationChannel())) {
+            if (StringUtils.isBlank(body)) {
+                String errorCode =
+                        I18nEmailUtil.prependOperationScenarioToErrorCode(
+                                TemplateMgtConstants.ErrorMessages.ERROR_CODE_INVALID_SMS_TEMPLATE.getCode(),
+                                TemplateMgtConstants.ErrorScenarios.NOTIFICATION_TEMPLATE_MANAGER);
+                throw new NotificationTemplateManagerClientException(errorCode,
+                        TemplateMgtConstants.ErrorMessages.ERROR_CODE_INVALID_SMS_TEMPLATE.getMessage());
+            }
+            if (StringUtils.isNotBlank(subject) || StringUtils.isNotBlank(footer)) {
+                String errorCode =
+                        I18nEmailUtil.prependOperationScenarioToErrorCode(
+                                TemplateMgtConstants.ErrorMessages.ERROR_CODE_INVALID_SMS_TEMPLATE_CONTENT.getCode(),
+                                TemplateMgtConstants.ErrorScenarios.NOTIFICATION_TEMPLATE_MANAGER);
+                throw new NotificationTemplateManagerClientException(errorCode,
+                        TemplateMgtConstants.ErrorMessages.ERROR_CODE_INVALID_SMS_TEMPLATE_CONTENT.getMessage());
+            }
+        } else {
+            if (StringUtils.isBlank(subject) || StringUtils.isBlank(body)) {
+                String errorCode =
+                        I18nEmailUtil.prependOperationScenarioToErrorCode(
+                                TemplateMgtConstants.ErrorMessages.ERROR_CODE_INVALID_EMAIL_TEMPLATE.getCode(),
+                                TemplateMgtConstants.ErrorScenarios.NOTIFICATION_TEMPLATE_MANAGER);
+                throw new NotificationTemplateManagerClientException(errorCode,
+                        TemplateMgtConstants.ErrorMessages.ERROR_CODE_INVALID_EMAIL_TEMPLATE.getMessage());
+            }
+        }
+    }
+
+    private String getDefaultNotificationLocale(String notificationChannel) {
+
+        if (NotificationChannels.SMS_CHANNEL.getChannelType().equals(notificationChannel)) {
+            return DEFAULT_SMS_NOTIFICATION_LOCALE;
+        } else {
+            return DEFAULT_EMAIL_NOTIFICATION_LOCALE;
+        }
+    }
+
+    private void assertSystemTemplateTypeExists(String templateType, String notificationChannel)
```

Comment:
rename all assert function to better function names

Reply:
addressed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/250/commits/4a4dae818d530081e8a91a33d6d70da13c375c9c